From 18c7e94edf6d13a39efa8e0e5485428d9920858b Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Sun, 2 Aug 2015 20:16:09 -0700
Subject: [PATCH 014/206] SnapdragonCamera: Re-enable ZSL after exiting HDR
 mode

* Make sure to check that HDR was changed before forcing ZSL back on.

Change-Id: Ifb76263c3a50b919762fd2a313d6013295c0619f
---
 src/com/android/camera/PhotoModule.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index 7d8e7efb2..fa0ec987c 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -4889,6 +4889,7 @@ public class PhotoModule
 
         //filter off unsupported settings
         final String settingOff = mActivity.getString(R.string.setting_off_value);
+        final String settingOn = mActivity.getString(R.string.setting_on_value);
         if (!CameraSettings.isZSLHDRSupported(mParameters)) {
             //HDR internally uses AE-bracketing. Disable both if not supported.
             if (notSame(pref, CameraSettings.KEY_CAMERA_HDR, settingOff) ||
@@ -4897,6 +4898,9 @@ public class PhotoModule
             } else if (notSame(pref,CameraSettings.KEY_ZSL,settingOff)) {
                 mUI.setPreference(CameraSettings.KEY_CAMERA_HDR, settingOff);
                 mUI.setPreference(CameraSettings.KEY_AE_BRACKET_HDR, settingOff);
+            } else if (notSame(pref, CameraSettings.KEY_CAMERA_HDR, settingOn) ||
+                notSame(pref, CameraSettings.KEY_AE_BRACKET_HDR, settingOn)) {
+                mUI.setPreference(CameraSettings.KEY_ZSL,settingOn);
             }
         }
 
-- 
2.17.1

